<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.12.1 by Michael Rose
  Copyright 2013-2018 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE.txt
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>HEGD - bioX-UPRM Workspace</title>
<meta name="description" content="An amazing website.">



<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="bioX-UPRM Workspace">
<meta property="og:title" content="HEGD">
<meta property="og:url" content="http://localhost:4000/_pages/HEGD/">












  

  


<link rel="canonical" href="http://localhost:4000/_pages/HEGD/">







  <script type="application/ld+json">
    {
      "@context": "http://schema.org",
      "@type": "Organization",
      "name": "bioX-UPRM Workspace",
      "url": "http://localhost:4000",
      "sameAs": null
    }
  </script>







<!-- end _includes/seo.html -->


<link href="/feed.xml" type="application/atom+xml" rel="alternate" title="bioX-UPRM Workspace Feed">

<!-- http://t.co/dKP3o1e -->
<meta name="HandheldFriendly" content="True">
<meta name="MobileOptimized" content="320">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">

<!--[if lte IE 9]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->


    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->


<link rel="apple-touch-icon" sizes="180x180" href="/assets/images/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/assets/images/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/assets/images/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">





<!-- end custom head snippets -->

  </head>

  <body class="layout--single wide">

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <a class="site-title" href="http://localhost:4000/"><img src="assets/images/bioXLogo.png"></a>
        <ul class="visible-links"><li class="masthead__menu-item">
              <a href="/Courses/" >Courses</a>
            </li><li class="masthead__menu-item">
              <a href="/Projects/" >Projects</a>
            </li><li class="masthead__menu-item">
              <a href="/Publications/" >Publications</a>
            </li></ul>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">
  
  <div class="sidebar sticky">
  


<div itemscope itemtype="http://schema.org/Person">

  
    <div class="author__avatar">
      

      
        <img src="/assets/images/JoseDeDiego.jpg" alt="" itemprop="image">
      
    </div>
  

  <div class="author__content">
    
      <h3 class="author__name" itemprop="name"></h3>
    
    
      <p class="author__bio" itemprop="description">
        Biomedical research BY students FOR students. HQ | Makerspace @ UPRM
      </p>
    
  </div>

  <div class="author__urls-wrapper">
    <button class="btn btn--inverse">Follow</button>
    <ul class="author__urls social-icons">
      
        <li itemprop="homeLocation" itemscope itemtype="http://schema.org/Place">
          <i class="fas fa-fw fa-map-marker-alt" aria-hidden="true"></i> <span itemprop="name">Mayag√ºez, PR</span>
        </li>
      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      

      <!--
  <li>
    <a href="http://link-to-whatever-social-network.com/user/" itemprop="sameAs">
      <i class="fas fa-fw" aria-hidden="true"></i> Custom Social Profile Link
    </a>
  </li>
-->
    </ul>
  </div>
</div>

  
  </div>


  <article class="page" itemscope itemtype="http://schema.org/CreativeWork">
    <meta itemprop="headline" content="HEGD">
    
    
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">HEGD
</h1>
          
        </header>
      

      <section class="page__content" itemprop="text">
        
        <h1 id="homology-based-essential-gene-discovery-tool">Homology-based Essential Gene Discovery Tool.</h1>

<h2 id="what-are-essential-genes">What are essential genes?</h2>

<p><strong>Essential genes</strong> are genes that are thought to be necessary for the survival of an organism. All organisms share groups of <strong>essential genes</strong> involved in processess associated to DNA Replication, Transcription, Translation, and Metabolism. However, species have developed different sub-sets of essential genes throughout the span of millions of years of evolution. Essential genes are often identified throughout transposon-mediated mutagenesis studies. In these types of studies, transposons - which are mobile genetic elements - are randomly inserted in as many positions in a genome as possible, and cells that survive after a given cell culture period are sequenced. The genes of surviving cells are screened for transposon-mediated mutations: it is expected that essential genes fail to accumulate such mutations, given that cells would not tolerate them if the genes affected were truly essential.</p>

<p><img src="../Imgs/Transposon_Essential.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 1: Identification of essential genes via genome-wide screen.</strong> Genomic regions corresponding to the genes hemA, hemC, hemD, and hemB failed to accumulate insertions of transposons - making said genes potentially essential. Source: <a href="https://en.wikipedia.org/wiki/Essential_gene">Wikipedia</a></p>
</blockquote>

<p>These types of studies require a lot of time an effort. Instead, we suggest implementing a homology-based machine learning model based on previous work <strong>[<a href="https://www.hindawi.com/journals/bmri/2016/7639397/#B28">1</a>]</strong> for the prediction of species-specific essential genes. <em>This strategy assumes that essential genes are conserved amongst multiple species per kingdom, given their biological relevance.</em></p>

<h2 id="project">Project.</h2>

<p><strong>Objective:</strong> Use the protein sequences of known essential genes in multiple species to predict essential genes in a previously unassessed species/strain.</p>

<p>The figure below only speaks well to trained data scientists. As a novice, you will need to continue reading this document in order to fully understand the whole approach and what your role will be in its implementation.</p>

<p><img src="../Imgs/HEGD_Concept.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 2: Concept behind HEGD.</strong> <em>N</em> number of models are created through reciprocal Blastp alignments between the proteins of <em>N</em> species with the goal of identifying <strong>orthologs</strong> - genes in different species that evolved from a common ancestral gene by speciation - of known essential genes. Models are then trained and tested using a Support Vector Machine (a type of supervised machine learning model). Multiple sub-sets of each model are generated through feature selection prior to performing cross validation. Once the models are properly trained and tested, users can then input the proteome of previously unassessed species, and a trained model is selected based on <strong>evolutionary distance</strong> - a number that defines how closely related a species is to another - with the purpose of classifying protein-coding genes as essential or non-essential. We have highlighted in Cyan the tasks that have been completed by Project Supervisors, and in Purple the tasks that will be implemented by the Students attending this course. To understand the HEGD approach in detail, please continue reading.</p>
</blockquote>

<ul>
  <li><strong>Reciprocal Blastp.</strong>
<img src="../Imgs/ReciprocalAlignment.png" alt="Alt Text" />
    <blockquote>
      <p><strong>Figure 3: Procedure for reciprocal blastp alignments.</strong> Given two species, A (Query) and B (Reference), each protein product in Species A (we will refer to each individual protein product as <em>Protein X</em>) is aligned with blastp to the whole proteome of Species B, and the best hit (alignment) is collected. The protein product pertaining the best hit in Species B is aligned to the whole Proteome of Species A, and the best hit is collected. If the best hit in Species A is equivalent to <em>Protein X</em>, then the protein hit in Species B is considered an ortholog of <em>Protein X</em>. By using a database of known essential genes per species, like the <a href="http://www.essentialgene.org/">Database of Essential Genes (DEG)</a>, we can classify orthologs of <em>Protein X</em> as <strong>essential orthologs</strong>. These steps are repeated until the whole proteome of Species A is assessed. Multiple reciprocal blastp alignments are performed for every reference species (i.e. species that possess a list of experimentally determined essential genes) - see <em>Model Creation</em>.</p>
    </blockquote>
  </li>
</ul>

<p>Result of a single reciprocal blastp alignment (Species A &lt; - &gt; Species B):</p>
<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Gene         Essential_Ortholog
Gene_1       1                       &lt;-Yes
Gene_2       0                       &lt;-No
Gene_3       0                       &lt;-No
...
Gene_Z       1                       &lt;-Yes
</code></pre></div></div>

<ul>
  <li><strong>Model creation.</strong>
<img src="../Imgs/ModelCreation.png" alt="Alt Text" />
    <blockquote>
      <p><strong>Figure 4: Creation of a model per reference species based on multiple reciprocal blastp alignments.</strong> For every species in N Reference species, N-1 reciprocal blastp alignments are performed against the remaining Reference Species in order to identify essential orthologs. This yields datasets with N-1 Binary features, which can be used to train and test machine learning models like Support Vector Machines.</p>
    </blockquote>
  </li>
</ul>

<p>Result of multiple reciprocal blastp alignments per species model:</p>
<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Model 1 (Species 1)   |     Species 2    Species 3      ...     Species N
Gene_1                |         1           0                       1
Gene_2                |         0           0                       0
...                   |        ...         ...                     ...
Gene_Z                |         1           1                       1

Model 2 (Species 2)   |     Species 1    Species 3      ...     Species N
Gene_1                |         1           1                       1
Gene_2                |         0           0                       1
...                   |        ...         ...                     ...
Gene_Z                |         1           1                       0

.
.
.

Model N (Species N)   |     Species 1    Species 2      ...      Species N-1
Gene_1                |         0           1                       1
Gene_2                |         0           0                       0
...                   |        ...         ...                     ...
Gene_Z                |         1           0                       0

</code></pre></div></div>

<h2 id="community-tasks">Community tasks.</h2>

<p>This section contains the theoretical knowledge pertaining tasks that each work group will be completing. All Python code involving said tasks will be discussed and implemented within bioXJMB‚Äôs Slack work group (see below).</p>

<h3 id="group-1-evolutionary-distance-of-two-species-by-proteome-comparison">Group 1: Evolutionary distance of two species by proteome comparison.</h3>
<p>In order to classify genes of species that have no experimentally determined essential genes, a single trained model from the N reference models available needs to be selected. The question then remains as to which. We aim to use evolutionary distance as a metric to select for said trained model, as classification accuracy is expected to increase as evolutionary distance between species decreases.</p>

<p><img src="../Imgs/ModelSelection.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 5: Model selection based on evolutionary distance.</strong> Species Q is a species that does not have a set of experimentally determined essential genes. One of the trained models from N reference models must be chosen in order to classify the genes of Species Q as essential or non-essential. The model with the shortest evolutionary distance between Species Q and each trained reference model (Model 3, in this case) is chosen.</p>
</blockquote>

<ul>
  <li><strong>Calculating evolutionary distance.</strong>
    <blockquote>
      <p>Many methods have been developed to calculate evolutionary distance, but we are only interested in implementing a computationally inexpensive way to compare the relatedness between all protein products found in two Species, A and B. <strong>To acheive this, we can treat amino acid sequences as strings of length k, or <em>k-strings</em>, where each protein product is constructed from multiple k-strings</strong> <strong>[<a href="https://link.springer.com/article/10.1007%2Fs00239-003-2493-7">2</a>]</strong>.</p>
    </blockquote>
  </li>
</ul>

<p>To explain how this notation is pertinent to estimating relatedness of the proteome of two species, consider the following:</p>
<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Let k = length of k-string

Given that there are 20 amino acids found in most Earth-bound species, we can assume that there are 
20^k possible amino acid combinations for any given k-string.

Example: If k = 2, 20^k = 20^2 = 400, and so there are 400 possible amino acid combinations for any 
given k-string.
i.e.     amino acids = ["A","C","D","E","F","G","H","I", "L","K","M","N","P","Q","R","S","T","V","W","Y"]
         k-string combinations = ["AA","AC","AD","AE","AF","AG","AH","AI", "AL","AK","AM","AN","AP","AQ",
         "AR","AS","AT","AV","AW","AY", ... , "YA","YC","YD","YE","YF","YG","YH","YI", "YL","YK","YM",
         "YN","YP","YQ","YR","YS","YT","YV","YW","YY"] &lt;- 400 amino acid combinations
         
By fragmenting individual proteins into k-strings, we can search for and report all possible amino acid 
k-string combinations within the proteome of a species. 

Let Protein X = "MYKCYK", and L = length of Protein X

There are (L - K + 1) possible k-string fragmentation patterns of length k in Protein X.
i.e. if k = 2, 
     Protein X Fragments = ["MY","YK","KC","CY","YK"] &lt;- (L-K+1) = (6-2+1) = 5 fragmentation patterns
     
     - The frecuency of the k-string "AA" in Protein X Fragments is 0.
     - The frecuency of the k-string "YK" in Protein X Fragments is 2.

The frecuency of each possible amino acid k-string combination in the proteome of a species can be used to 
calculate probabilities of k-string occurrence. Said probabilities can be implemented in order to calculate 
k-string correlations between two species and, ultimately, evolutionary distance. 
</code></pre></div></div>

<p><img src="../Imgs/Probab_kstring.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 6: Probability of finding a particular k-string in a given protein sequence.</strong> The probability of finding a particular k-string (Œ±<sub>1</sub>Œ±<sub>2</sub>‚Ä¶Œ±<sub>k</sub>, where Œ± = amino acid) in a given protein sequence - say Protein X - results from the frecuency of said k-string in Protein X, divided by the amount of possible k-string fragmentation patterns of length k in Protein X (i.e. Length of Protein X - k + 1). Each individual probability per protein can be added up to yield the probability of finding a particular k-string in <em>ANY</em> given protein sequence of the species in question.</p>
</blockquote>

<p><img src="../Imgs/componentFormula.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 7: Emphasizing selective diversification of protein sequences.</strong> Neutral mutations (mutations that have no impact on biological fitness) lead to randomness in the k-string composition of proteins. In order to emphasize the selective diversification of sequence composition, we can subtract a random background, in the form of a predicted or inferred probability (P<sub>0</sub>), from the probability obtained via k-string frecuency assessment.</p>
</blockquote>

<p><img src="../Imgs/PredictedProbab_Kstring.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 8: Predicted Markovian Probability (P<sup>0</sup>) of finding a particular k-string in a given protein sequence.</strong> A widely used Markov model <strong>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/3078230">3</a>]</strong> can calculate a predicted probability that accounts for the random background discussed in the description of <strong>Figure 7</strong>.</p>
</blockquote>

<p><img src="../Imgs/CompAB.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 9: Representative vectors for Species A and B.</strong> Each representative vector possesses a Y-dimensional space, where Y=20<sup>k</sup>, of composition vectors. Representative vectors A and B store Y results (i.e. one for each possible k-string) from the formula portrayed in <strong>Figure 7</strong>.</p>
</blockquote>

<p><img src="../Imgs/CorrelationA_B.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 10: Correlation of Species A and B, defined by the cosine function of the angle between the two representative vectors, A &amp; B.</strong> The correlation of Species A and B actually refers to their proteomic distance, which resides in the interval (-1,1).</p>
</blockquote>

<p><img src="../Imgs/DistanceA_B.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 11: Normalization of correlation between Species A and Species B yields evolutionary distance.</strong> Because the cosine function defining the correlation of Species A and B results in values between -1 and 1, we can use this formula to normalize values to the interval (0,1).</p>
</blockquote>

<h3 id="group-2-training-and-testing-a-support-vector-machine-for-the-classification-of-genes-as-essential-or-non-essential">Group 2: Training and testing a Support Vector Machine for the classification of genes as essential or non essential.</h3>

<p>Support Vector Machines (SVM) are one of the most widely used machine learning approaches when it comes to Biological Data. Each data point in an SVM is viewed as a p-dimensional vector, where such points are expected to be able to linearly separate with a (p-1)-dimensional hyperplane. Such a hyperplane is referred to as the optimal hyperplane, and decision boundaries are generated with the help of support vectors - vectors that greatly contribute to the identification of an optimal hyperplane.</p>

<p><img src="../Imgs/SVM_Concept.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 12: Training VS Classifying/Testing in a two-dimensional SVM.</strong> SVM models are first trained using a training dataset that contains two features (value columns) or more and a class label (+ or -). Each sample (row) contributes to the identification of an optimal hyperplane, which creates the decision boundaries that dictate classification results. Trained models can be tested using testing datasets that share the same features, but not the same samples. Testing sets also possess known labels. When testing trained models, each testing sample is assigned a label based on the decision boundaries established throughout training; assigned labels are compared to known labels and an accuracy score is used to rank the performance of the model. Once a satisfactory threshold is reached, trained models can be used to infer the labels of unclassified datasets.</p>
</blockquote>

<p>We will train Support Vector Machines using multiple model-subsets derived from the N models created from each reference species, and we will test each model sub-set through cross-validation in order to keep the most accurate one (see <strong>Figure 13</strong> below).</p>

<ul>
  <li><strong>Feature Ranking.</strong>
<img src="../Imgs/FeatureRanking.png" alt="Alt Text" />
    <blockquote>
      <p><strong>Figure 13: Using scoring methods to rank features while training and testing.</strong> While datasets with N-1 features resulting from multiple reciprocal blastp alignments can be used to train and test machine learning models (see <strong>Figure 4</strong>), scoring functions can be applied to rank features and create model sub-sets that are in turn trained and tested in order to save the model sub-set with the highest classification accuracy.</p>
    </blockquote>
  </li>
</ul>

<p><img src="../Imgs/F_Score.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 14: F score function.</strong> Given that our models contain N-1 binary features, we can use the F score function <strong>[<a href="https://www.csie.ntu.edu.tw/~cjlin/papers/features.pdf">4</a>]</strong> to rank features based on their ability to tell a positive sample from a negative sample. <strong>x<sub>i</sub></strong> , <strong>x<sub>i</sub><sup>(+)</sup></strong>, and <strong>x<sub>i</sub><sup>(-)</sup></strong> are the mean of the i<sub>th</sub> feature of the total, positive, and negative samples respectively. x<sub>k,i</sub><sup>(+)</sup> is the i<sub>th</sub> feature of the k<sub>th</sub> postive sample, while x<sub>k,i</sub><sup>(-)</sup> is the i<sub>th</sub> feature of the k<sub>th</sub> negative sample.</p>
</blockquote>

<ul>
  <li><strong>Model training.</strong>
    <blockquote>
      <p>We mentioned earlier that data points of a training set are often expected to be linearly separable. However, this is not always the case. In order to properly separate datapoints, we can implement kernel functions with the purpose of mapping the original finite-dimensional space into a higher-dimensional space. We will be implementing a Gaussian kernel function in <a href="http://scikit-learn.org/stable/modules/svm.html">Sci-Kit Learn</a> (a ML Library in Python).</p>
    </blockquote>
  </li>
  <li><strong>Model testing.</strong>
    <blockquote>
      <p>We will be performing 10-fold cross-validation in order to test each one of our classification models.</p>
    </blockquote>
  </li>
</ul>

<p><img src="../Imgs/CrossValidation.png" alt="Alt Text" /></p>
<blockquote>
  <p><strong>Figure 15: Schematic of 10-fold cross-validation.</strong> The training data is randomly sorted and divided into 10 equal portions, where 9 portions are used to train the classification model, and 1 portion is used as a testing set. The process is repeated until all portions are used as testing sets. 10-fold cross-validation follows the Pareto principle, which states that in most events at least 80% of the effects come from 20% of the causes.</p>
</blockquote>

<h3 id="all-work-will-be-carried-out-and-monitored-through-the-bioxjmb-slack-group"><strong>All work will be carried out and monitored through the bioXJMB Slack group.</strong></h3>

<p><a href="https://join.slack.com/t/bioxjmb/shared_invite/enQtNDIwNDM0MDA2NzkwLTZjMWFkYmI5YjBjZmVlYmM0MDAzNzYyMGUyNDAwNmNjN2JjYzVlMjRiNzE1NGNhYTMzOWY5MzU2MGYwNmFhYjE" class="btn btn--info btn--large"><i class="fab fa-slack"></i> @bioXJMB</a></p>

<hr />
<h3 id="references">References:</h3>
<p>[<a href="https://www.hindawi.com/journals/bmri/2016/7639397/#B28">1</a>] Hong-Li Hua, et al. (2016). An Approach for Predicting Essential Genes Using Multiple Homology Mapping and Machine Learning Algorithms. <em>BioMed Research International</em>. Volume 2016, Article ID 7639397.</p>

<p>[<a href="https://link.springer.com/article/10.1007%2Fs00239-003-2493-7">2</a>] Qi Ji et al. (2003). Whole Proteome Prokaryote Phylogeny Without Sequence Alignment: A K-String Composition Approach. <em>J Mol Evol</em>. 58:1‚Äì11.</p>

<p>[<a href="https://www.ncbi.nlm.nih.gov/pubmed/3078230">3</a>] Brendel V, et al. (1986) Linguistics of nucleotide sequences: Morphology and comparison of vocabularies. J Biomol Struct Dyn 4:11‚Äì21</p>

<p>[<a href="https://www.csie.ntu.edu.tw/~cjlin/papers/features.pdf">4</a>] Y.-W. Chen &amp; C.-J Lin. (2006). Combining SVMs with various feature selection strategies. <em>Studies in Fuzziness and Soft Computing</em>. 207:315‚Äì324.</p>

<h4 id="project-coordinator">Project Coordinator:</h4>
<ul>
  <li><a href="https://github.com/CharlesSanfiorenzo/Bioinformatics">Charles Sanfiorenzo</a> - csanfior@caltech.edu</li>
</ul>


        
      </section>

      <footer class="page__meta">
        
        


        
      </footer>

      

      
    </div>

    
  </article>

  
  
</div>
    </div>

    

    <div class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2018 bioX-UPRM Workspace.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>
  <script src="https://use.fontawesome.com/releases/v5.2.0/js/all.js"></script>








  </body>
</html>